test/parallel/test-async-hooks-async-await-regression.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const common = require('../common');
+
+const async_hooks = require('async_hooks');
+const assert = require('assert');
+
+common.crashOnUnhandledRejection();
+
+const asyncIds = [];
+
+async_hooks.createHook({
+  init: (asyncId, type, triggerAsyncId, resource) => {
+    asyncIds.push(`${triggerAsyncId} => ${asyncId}`);
+  }
+}).enable();
+
+async function main() {
+  console.log('hello');
+  await null;
+  console.log('world');
+}
+
+main().then(() => assert.deepStrictEqual(
+  asyncIds, [ '1 => 3', '1 => 4', '1 => 5', '3 => 6', '3 => 7' ]));
```

Comment:
I fear these precise numbers would cause brittleness going forward. The relationships between the numbers matters but not their precise values.